gendiff/diff_nested.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, Dict
+
+from gendiff.diff import get_diff_view
+from gendiff.formatting import convert_value, nested_diff, plain_diff
+
+
+def get_diff_str_nested(
+    dict1: Dict[str, Any], dict2: Dict[str, Any], indent: int = 2
+) -> str:
+    diff_view = get_diff_view(dict1, dict2)
+    diff_str = "{\n" if indent == 2 else ""
+
+    for key, val in diff_view.items():
+        value = convert_value(val["value"])
+        ch_value = convert_value(val["changed_value"])
+        whitespace_or_not = "" if not value else " "
+
+        if val["type"].endswith("nested"):
+            if val["type"] == "nested":
+                key_type = "nested"
```

Comment:
[optional] key_type is the same as `val["type']` (most of the time), consider using it instead